SetTheory/set-relation.R

```diff
@@ -293,175 +293,187 @@ getAsymDiffMC <- function(numSets=2, setSize=5, dType = 1, difficulty = 1) {
-  answer <- not(sourceSet, universalSet)
+  
+  distractors <- vector(mode="list", length = 3)
   
   #distractor 1 is the intersection of the sets which is the wrong answer
-  d1 <- intersect(sourceSet, universalSet)
+  d1 <- replace(d1, length(d1) - 2, getValue(x = dType, min = 1, max = 20, cat = 6)) 
   
   #distractor 2 is the intersection and the complement together which is wrong
-  d2 <- universalSet
+  d2 <- replace(d2, length(d2), getValue(x = dType, min = 1, max = 20, cat = 6))
   
   #distractor 3 is the original set which is not the complement and is wrong
-  d3 <- sourceSet
+  d3 <- sourceSets[[1]]
   
-  wrongs <- list()
-  wrongs[[1]] <- d1
-  wrongs[[2]] <- d2
-  wrongs[[3]] <- d3
+  distractors[[1]] <- formatListAsSet(d1)
+  distractors[[2]] <- formatListAsSet(d2)
+  distractors[[3]] <- formatListAsSet(d3)
   
-  #format wrong answers as strings
+  
+  
+  #Iterate through the sourceSets. format list as Set and insert at the index.
   counter <- 1
-  for (w in wrongs){
-    current <- formatListAsSet(w)
-    wrongs[counter] <- current
+  for (s in sourceSets){
+    sourceSets[counter] <- formatListAsSet(s)
     counter <- counter + 1
   }
   
-  #format source set as string
-  current <- formatListAsSet(sourceSet)
-  
-  
-  answer <- formatListAsSet(answer)
-  
-  #inserting string formatting "Let A = ...."
-  sourceSet <- insertSetQStrings(sourceSet)
-  
-  #The actual question being asked of the sets. 
-  questionStr <- "Let A be a single set. What is the complement of set A?"
-  
-  #combining the source sets and question string.
-  sourceSet <- c(questionStr, sourceSet)
+  #format the the sourceSets as Question Strings
+  # "A = {...}"
+  # "B = {...}"
+  sourceSets <- insertSetQStrings(sourceSets)
   
-  #format answers and sources into json and return results 
-  toSend <- list(content= sourceSet, correct= answer, distractors= wrongs)
+  # now we concatenate the question contents together
+  questionContents <- c(questionText, sourceSets)
   
+  #add all items to a list for return
+  toSend <- list(content = questionContents, correct = correct, distractors = distractors)
   
+  #return question info
   return(toSend)
-  
 }
 
-getSetEqualityMC <- function(n=2, m = 5, dType = 1) {
-  n <-2   #currently, the api only supports 2 sets. 
   
-  sourceSets <- list()    #creating an empty list. Elements will be sets 
+# getSetEquality() generates and prepares 2 sets of m members 
+# as well as 1 "distractor" and 1 correct answer.
+# The correct answer is a string which states whether the sets are equal
+# or not.
+#
+#
+# @param  numSets        The number of sets to consider
+# @param  setSize        The number of elements in each set. 
+# @response json         A json object containing the
+#                        sets, correct, and incorrect
+#                        answers.
+#
+
+getSetEqualityMC <- function(numSets = 2, setSize = 5, dType = 1) {
+  questionText <- "Let A and B be two sets. Are A and B equal?"
   
```

Comment:
It appears that this question is designed to **only** consider 2 sourceSets. That's fine, but you need to hard-code that into the function. Currently, the default value is defined in line 378 "...numSets = 2, ...". But that's just the default. You should declare "numSets <-2" on line 380. That way any other value passed to the function for numSets is overridden.